### PR TITLE
Paritionable model with grouped resources

### PR DESCRIPTION
**Pull request by johnbelamaric**
This is another alternative to the partitionable model.

The PR is a delta on top of #27.

Rather than flattening the individual physical GPU named resources into a single array, this is closer to Kevin's original, and keeps each physical GPU's shared resources in a separate struct. The individual devices then consume from those structs, by name. It is also possible for a device to consume from multiple shared resource groups, though that is not shown in the example.

### CODE REVIEW

File: dra-evolution/pkg/api/capacity_types.go [hunk 1]
```diff
@@ -69,6 +78,25 @@ type ResourcePoolSpec struct {
...
+	Name string `json:"name"`
+
+	// SharedCapacity defines the sets of shared capacity consumable by
+	// devices in this SharedCapacityGroup.
+	//
+	// Must not have more than 16 entries.
+	//
+	// +listType=atomic
+	// +optional
+	SharedCapacity []SharedCapacity `json:"sharedCapacity,omitempty"`
```
Comment (klueska) [inline dra-evolution/pkg/api/capacity_types.go:96]: Maybe `items` is a better name so it's not stuttering so much.
```suggestion
	Items []SharedCapacity `json:"items,omitempty"`
```
I'd also say that its not optional. If you have one of these, it should have at least one item in it.

File: dra-evolution/pkg/api/capacity_types.go [hunk 2]
```diff
@@ -59,6 +59,15 @@ type ResourcePoolSpec struct {
...
 
+	// SharedCapacity defines the sets of shared capacity consumable by
+	// devices in this ResourceSlice.
+	//
+	// Must not have more than 16 entries.
+	//
+	// +listType=atomic
+	// +optional
+	SharedCapacity []SharedCapacityGroup `json:"sharedCapacity,omitempty"`
+
```
Comment (klueska) [inline dra-evolution/pkg/api/capacity_types.go:70]: Does it make sense to double down on this PR and add the notion of `SharedAttributes` with a similar structure (i.e. a named group of attributes)? Then we can create groups of common attributes that individual devices can reference by name to include (in addition to their own local attributes, which would take precedence on nameing conflicts).

Comment (klueska) [reply dra-evolution/pkg/api/capacity_types.go:70]: I had this in some proposal somewhere, but I an't seem to find it anymore.

Comment (johnbelamaric) [reply dra-evolution/pkg/api/capacity_types.go:70]: That's combining this with option 3. I am working through various options, see https://github.com/kubernetes-sigs/wg-device-management/issues/20

Let's see all the options before we decide which to iterate on.

**Comment by thockin** [comment]
Since this is "what's in the KEP" I think we should merge it and rebase all the options on it, so they appear as diffs.

/lgtm

**Comment by thockin** [comment]
Shoot, I screwed up and approved the wrong one!  I meant to LGTM #27 !!!

**Pull request merged**